dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/outlier/data/OutlierRequest.java

```diff
@@ -0,0 +1,151 @@
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.analytics.outlier.data;
+
+import static org.hisp.dhis.analytics.OutlierDetectionAlgorithm.Z_SCORE;
+import static org.hisp.dhis.analytics.outlier.Order.MEAN_ABS_DEV;
+import static org.hisp.dhis.common.OrganisationUnitDescendants.DESCENDANTS;
+import static org.springframework.util.Assert.notNull;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import lombok.Getter;
+import org.hisp.dhis.analytics.OutlierDetectionAlgorithm;
+import org.hisp.dhis.analytics.outlier.Order;
+import org.hisp.dhis.common.OrganisationUnitDescendants;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+
+/**
+ * Encapsulation of an outlier detection request.
+ *
+ * @author Lars Helge Overland
+ */
+@Getter
+public class OutlierRequest {
+  private List<DataElement> dataElements = new ArrayList<>();
+
+  private Date startDate;
+
+  private Date endDate;
+
+  private List<OrganisationUnit> orgUnits = new ArrayList<>();
+
+  private OrganisationUnitDescendants orgUnitSelection;
+
+  private OutlierDetectionAlgorithm algorithm;
+
+  private double threshold;
+
+  private Date dataStartDate;
+
+  private Date dataEndDate;
+
+  private Order orderBy;
+
+  private int maxResults;
+
+  public List<Long> getDataElementIds() {
+    return dataElements.stream().map(DataElement::getId).toList();
+  }
+
+  private OutlierRequest() {}
+
+  public boolean hasDataStartEndDate() {
+    return dataStartDate != null && dataEndDate != null;
+  }
+
+  public static class Builder {
```

Comment:
lombok's ```@Builder``` would be preferrable rather than all this boilerplate code

Reply:
Done